src/pip/_internal/network/auth.py

```diff
@@ -145,10 +150,12 @@ def _get_new_credentials(self, original_url: str, allow_netrc: bool = True,
         # If we don't have a password and keyring is available, use it.
         if allow_keyring:
             # The index url is more specific than the netloc, so try it first
+            # fmt: off
             kr_auth = (
                 get_keyring_auth(index_url, username) or
                 get_keyring_auth(netloc, username)
             )
+            # fmt: on
```

Comment:
Ah dammit. I was explicitly trying to avoid mixing manual changes from automated ones. >.<

Anyway, removing this creates this patch:

```diff
diff --git a/src/pip/_internal/network/auth.py b/src/pip/_internal/network/auth.py
index 74d225472..0713f0205 100644
--- a/src/pip/_internal/network/auth.py
+++ b/src/pip/_internal/network/auth.py
@@ -150,12 +150,9 @@ def _get_new_credentials(
         # If we don't have a password and keyring is available, use it.
         if allow_keyring:
             # The index url is more specific than the netloc, so try it first
-            # fmt: off
-            kr_auth = (
-                get_keyring_auth(index_url, username) or
-                get_keyring_auth(netloc, username)
+            kr_auth = get_keyring_auth(index_url, username) or get_keyring_auth(
+                netloc, username
             )
-            # fmt: on
             if kr_auth:
                 logger.debug("Found credentials in keyring for %s", netloc)
                 return kr_auth
```

I think that's clearly worse. It's https://github.com/psf/black/issues/2156 upstream.